### PR TITLE
[fuzz]expand readfilter_fuzzer to cover mongo_proxy and mysql_proxy

### PR DESCRIPTION
**Pull request by jianwen612**
Signed-off-by: jianwen <jianwendong@google.com>

Additional Description:

Added coverage for mongo_proxy and mysql_proxy
Added test cases(corpus) for them.

Coverage report for those additional 2 filters(running for 60 seconds with 18 filters):
`Directory name`--------------------------------------------------------------------------------------/	`Line Coverage`       / 	`Functions coverage`
 `source/extensions/filters/network/mysql_proxy`----------/  `58.5 %`  `465 / 795`           /	`66.1 %`	`80 / 121`
`source/extensions/filters/network/mongo_proxy`---------/`33.4 %`	`624 / 1869`   /`35.9 %`	`80 / 223`

Risk Level:Low

/cc @asraa 
/cc @samkerner

### CODE REVIEW

File: test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc [hunk 1]
```diff
@@ -157,6 +161,20 @@ void UberFilterFuzzer::checkInvalidInputForFuzzer(const std::string& filter_name
           "http_conn_manager trying to use Quiche which we won't fuzz here. Config:\n{}",
           config.DebugString()));
     }
+  } else if (filter_name == NetworkFilterNames::get().MongoProxy) {
+    envoy::extensions::filters::network::mongo_proxy::v3::MongoProxy& config =
+        dynamic_cast<envoy::extensions::filters::network::mongo_proxy::v3::MongoProxy&>(
+            *config_message);
+    if (config.has_delay() && config.mutable_delay()->has_header_delay()) {
+      // MongoProxy filter doesn't allow header_delay because it will pass nullptr to percentage()
```
Comment (asraa) [inline test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc:169]: Is this field marked as unsupported in the API? Passing a `nullptr` to a function like this seems like a bug https://github.com/envoyproxy/envoy/blob/466245b50e6d475be841e1c8d5f254dd6ea41272/source/extensions/filters/common/fault/fault_config.cc#L15

Comment (jianwen612) [reply test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc:169]: > Is this field marked as unsupported in the API? Passing a `nullptr` to a function like this seems like a bug
> 
> https://github.com/envoyproxy/envoy/blob/466245b50e6d475be841e1c8d5f254dd6ea41272/source/extensions/filters/common/fault/fault_config.cc#L15

I talked with author of this line @mattklein123, and he agrees that this is a config issue, and mongo's delay should be a fixed_delay.  For fixed_delay, nullptr is ok. But for header_delay, nullptr is not allowed. I think they extend the same base class and thus one allows nullptr and one doesn't.

see class FixedDelayProvider (source/extensions/filters/common/fault/fault_config.h) and class HeaderDelayProvider (source/extensions/filters/common/fault/fault_config.h)

**Comment by jianwen612** [comment]
> Thanks! One quick question about the header_delay field...

I added a handle for nullptr in `HeaderPercentageProvider::percentage` in (https://github.com/envoyproxy/envoy/pull/12628)

I can removed these code after that PR get merged.

File: test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc (hunk 1 above)
Comment (asraa) [reply test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc:169]: +1 to opting for the solution here https://github.com/envoyproxy/envoy/pull/12628/files
which guards against derefencing a nullptr passing into `percentage`.

**Comment by jianwen612** [comment]
> Thanks! This LGTM -- let's merge https://github.com/envoyproxy/envoy/pull/12628/files first and remove th invalid input for header delay?

Yes. I remove these after it is merged.

File: test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc (hunk 1 above)
Comment (sqkerner) [reply test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc:169]: Other than this issue (which will be solved by committing  https://github.com/envoyproxy/envoy/pull/12628/), this change looks good to me.

**Comment by jianwen612** [comment]
> Thanks! This LGTM -- let's merge https://github.com/envoyproxy/envoy/pull/12628/files first and remove th invalid input for header delay?

Removed!

**Pull request merged**